registry/quilt_server/views.py

```diff
@@ -1617,6 +1618,53 @@ def list_users():
 
     return resp.json()
 
+@app.route('/api/users/list_detailed', methods=['GET'])
+@api(enabled=ENABLE_USER_ENDPOINTS, require_admin=True)
+@as_json
+def list_users_detailed():
+    package_counts_query = (
+        db.session.query(Package.owner, sa.func.count(Package.owner))
+        .group_by(Package.owner)
+        )
+    package_counts = dict(package_counts_query) 
+
+    events = (
+        db.session.query(Event.user, Event.type, sa.func.count(Event.type))
+        .filter(Event.type.in_([Event.Type.INSTALL, Event.Type.PREVIEW]))
+        .group_by(Event.user, Event.type)
+        )
+
+    event_results = defaultdict(lambda: defaultdict(int))
+    for event_user, event_type, event_count in events:
+        if event_type == Event.Type.INSTALL:
+            event_results[event_user]["installs"] = event_count
```

Comment:
Wait, you're not actually returning `"installs"` or `"previews"` to the user - you're just looking them up later in the code. So you don't need strings. Or nested dictionaries. Just do:
```
event_results[(event_user, event_type)] = event_count
```

In fact:
```
event_results = {
  (event_user, event_type): event_count
  for ...
}
```